### DISCUSSION TITLE
Relationship and property ownership

### DISCUSSION THREAD

**Discussion by mikekidner** [gaphor/gaphor#2339] [Q&A] [answered]
I've noticed that when creating ports on a part, the port does not fall below the block that "owns" the part in the model hierarchy. It tends to appear at the top level, this is also true to relationships in BDD or IBD. Is this behaviour intentional? I was expecting ports to "belong" to a block by default and similarly for relationships (the source block).

What is the intended behaviour of ports and relationships in terms of where they appear in the model hierarchy?

**Comment by danyeaw** [comment]
Hi @mikekidner I think you are right, those things should be properly nested. Would you be interested in submitting an issue?

**Reply by mikekidner** [reply, marked as answer]
Thanks for the response, I've created a new [issue](https://github.com/gaphor/gaphor/issues/2368)
